WebDriverAgentLib/Utilities/FBXCodeCompatibility.m

```diff
@@ -48,9 +41,12 @@ - (void)fb_terminate
   }
 }
 
-- (NSUInteger)fb_state
+- (void)fb_launch
 {
-  return [[self valueForKey:@"state"] intValue];
+  [self launch];
+  if (![self fb_waitForAppElement:APP_STATE_CHANGE_TIMEOUT]) {
```

Comment:
Not for this pr itself, but I wonder if we could remove this wait...

This is on tvOS. When `    t =    72.77s         Wait for com.apple.TVSettings to idle` was shown as below, then the new session request/activate app waited a couple of seconds then printed `The application 'com.apple.TVSettings' is not running in foreground after 5.00 seconds`.


```
The application 'com.apple.TVSettings' is not running in foreground after 5.00 seconds
    t =    67.35s Terminate com.apple.TVSettings:844

    t =    72.70s Open com.apple.TVSettings
    t =    72.70s     Launch com.apple.TVSettings
    t =    72.76s         Setting up automation session
Waiting up to 10s until com.apple.TVSettings is in idle state (including animations)
    t =    72.77s         Wait for com.apple.TVSettings to idle

The application 'com.apple.TVSettings' is not running in foreground after 5.00 seconds
```

But when I ran below app state check without `if (![self fb_waitForAppElement:APP_STATE_CHANGE_TIMEOUT]) {` condition check. Then, after starting a session with `com.apple.TVSettings` bundle id, then the session started after ` t =    72.77s         Wait for com.apple.TVSettings to idle`. Then, I immediately sent app state check as below:

```
[HTTP] --> POST /session/a9cf753b-b99a-4da6-9079-ee47061bfe91/appium/device/app_state
[HTTP] {"appId":"com.apple.TVSettings"}
[debug] [XCUITestDriver@084e (a9cf753b)] Calling AppiumDriver.queryAppState() with args: ["com.apple.TVSettings","a9cf753b-b99a-4da6-9079-ee47061bfe91"]
[debug] [XCUITestDriver@084e (a9cf753b)] Executing command 'queryAppState'
[debug] [XCUITestDriver@084e (a9cf753b)] Proxying [POST /wda/apps/state] to [POST http://192.168.5.0:8100/session/807C7B5E-1F64-46A6-9CC4-0390EC7169B4/wda/apps/state] with body: {"bundleId":"com.apple.TVSettings"}
[debug] [XCUITestDriver@084e (a9cf753b)] Got response with status 200: {"value":4,"sessionId":"807C7B5E-1F64-46A6-9CC4-0390EC7169B4"}
[debug] [XCUITestDriver@084e (a9cf753b)] Responding to client with driver.queryAppState() result: 4
[HTTP] <-- POST /session/a9cf753b-b99a-4da6-9079-ee47061bfe91/appium/device/app_state 200 20 ms - 11
```

Then `com.apple.TVSettings` was responded as foreground while the timing should have raised `The application 'com.apple.TVSettings' is not running in foreground after 5.00 seconds` message with the `if (![self fb_waitForAppElement:APP_STATE_CHANGE_TIMEOUT]) {` .

`http://192.168.5.0:8100/wda/activeAppInfo` endpoint also responded the launched app while `The application 'com.apple.TVSettings' is not running in foreground after 5.00 seconds` message was shown timing.

Maybe... we added this for some past reported issue... but I wondered if we could remove knowadays

Reply:
I mean without `![self fb_waitForAppElement:APP_STATE_CHANGE_TIMEOUT]`, the app interaction could started 3 sec or around after after a new session starts at that time.

Reply:
Oh, this is maybe tvOS only thing

Reply:
it could be these checks are redundant now since they have been added a while ago. Could you please check again without them?

Reply:
Tested. My env was ok with the latest removed one

Reply:
Checked in the past PRs. Perhaps https://github.com/appium/WebDriverAgent/pull/205 is the original one, which was for iOS 13 for https://github.com/appium/WebDriverAgent/pull/204 , so maybe actually we can remove this safely